src/transport/SessionManager.cpp

```diff
@@ -150,12 +150,15 @@ CHIP_ERROR SessionManager::PrepareMessage(const SessionHandle & sessionHandle, P
         auto * groups     = Credentials::GetGroupDataProvider();
         VerifyOrReturnError(nullptr != groups, CHIP_ERROR_INTERNAL);
 
+        FabricInfo * fabric = mFabricTable->FindFabricWithIndex(groupSession->GetFabricIndex());
+        VerifyOrDie(fabric != nullptr);
```

Comment:
I don't think we should die on this.  Group session can have been constructed with any fabric index, valid or not, and there is a clear path to return `CHIP_ERROR_INVALID_ARGUMENT` here.

The appropriateness of dying here I think depends in part on whether we can say for sure that it would have been the sdk itself that produced the invalid fabric index.  It is not clear to me that we can guarantee this wasn't injected from outside the sdk.

Reply:
done

Reply:
Thank you.